### ISSUE TITLE
2.4 Bug: Preferences reset when Kratos session expires.

### ISSUE THREAD

**Issue by zfcarsonb** [Security-Onion-Solutions/securityonion#11330]
## Description

It appears that preferences (e.g. dark mode, table sorting, fetch limits, etc.) reset to their defaults when a Kratos session expires.

## Environment

- Deployment:
  - Version: 2.4
  - Type: Distributed
  - Access via: FQDN
- Nodes:
  - 1x Manager
  - 1x Search
  - 1x Forward
  - 2x Receiver
  - 3x IDH

## Expected Behavior

Cookie or local storage-set preferences should be retained across sessions.

## Observed Behavior

Cookie or local storage-set preferences get reset to default values when the Kratos session expires.

## Other Notes

- This bug seems only to apply to 2.4. The bug was not present in 2.3.
- I am using WebAuthn for password-less authentication.

### Example 1

1. Log into Security Onion
2. Change theme to Light Mode
3. Wait until session expires
4. Log back in.
5. Theme resets to dark mode.

### Example 2

1. Log into Security Onion
2. Change alert tables to sort by `event.severity_label`
3. Wait until session expires
4. Log back in.
5. Table sorting resets to default.

**Comment by jertel** [comment]
I'm unable to reproduce this. I've tried Firefox, Chrome, Chromium, and Safari and all correctly preserve the local storage settings. 

I tried using password logins as well as passwordless logins.

I tried using a Macbook as well as a non Mac computer.

Kratos sessions depend on cookies, which do expire. However, settings like Dark mode are not tied to cookies, they're kept in local storage which does not expire. 

How long does it take for your "Kratos session" to timeout? Has that session lifespan been altered from the default?

I suggest reproducing this on a different machine and potentially a different network. If you're using a corporate issued computer, which has potentially enterprise-specific customizations and/or settings configured into the browser then you will need to try a computer and/or browser that is not compromised by those changes made by your IT department. If your network forces all traffic through a proxy, try using a system that can bypass the proxy. Use these steps to isolate why this is happening to you but not others in the community.

You can also inspect the browser's local storage before the session times out and after, to see if the settings disappear from the storage list. They should not.

**Comment by zfcarsonb** [comment]
Hi @jertel, thanks for taking the time to look into this.

I believe I have discovered the issue, based on your suggestion to check local storage. When we upgraded from Security Onion 2.3 to 2.4.10, we performed a full wipe and clean installation. However, we did not clear our browser's cookies and local storage from the previous 2.3 installation. This appeared caused issues when SOC tried to read or write local storage. Removing all browser artifacts from the 2.3 installation resolved the issue. I could be wrong about this, but I am no longer seeing the issue now.

Before the above discovery, I also looked at the Kratos salt pillar, and the default session expiry time has not been modified from 24 hours. I had also tried temporarily disabling browser group policy templates per your suggestion.

Thanks again for your help!
